sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2565,15 +2565,15 @@ case class ToBinary(expr: Expression, format: Option[Expression], child: Express
       case lit if lit.foldable =>
         val value = lit.eval()
         if (value == null) Literal(null, BinaryType)
-        else {
+        else if (value.isInstanceOf[UTF8String]) {
```

Comment:
Is it possible for `value` to be an instance of `java.lang.String`?

Reply:
no, it can only be `UTF8String`

Reply:
seems we can remove this check now, as we have checked the data type ahead

Reply:
Makes sense! Removed.